Source/WebCore/dom/Document.cpp

```diff
@@ -8933,6 +8933,66 @@ HTMLDialogElement* Document::activeModalDialog() const
     return nullptr;
 }
 
+HTMLElement* Document::topmostAutoPopover() const
+{
+    for (auto& element : makeReversedRange(m_topLayerElements)) {
+        if (auto* candidate = downcast<HTMLElement>(element.ptr()); candidate->popoverState() == PopoverState::Auto)
+            return candidate;
+    }
+
+    return nullptr;
+}
+
+// https://html.spec.whatwg.org/#hide-all-popovers-until
+void Document::hideAllPopoversUntil(Element* endpoint, FocusPreviousElement focusPreviousElement, FireEvents fireEvents)
+{
+    Vector<Ref<HTMLElement>> popoversToHide;
+    for (auto& element : makeReversedRange(m_topLayerElements)) {
+        if (!is<HTMLElement>(element) || downcast<HTMLElement>(element.get()).popoverState() != PopoverState::Auto)
+            continue;
+        if (element.ptr() == endpoint)
+            break;
+        popoversToHide.append(downcast<HTMLElement>(element.get()));
+    }
+
+    for (auto& popover : popoversToHide)
+        popover->hidePopoverInternal(focusPreviousElement, fireEvents);
+}
+
+void Document::handlePopoverLightDismiss(Event& event)
+{
+    ASSERT(event.isTrusted());
+
+    RefPtr topmostAutoPopover = this->topmostAutoPopover();
+    if (!topmostAutoPopover)
+        return;
+
+    RefPtr popoverToAvoidHiding = [&]() -> HTMLElement* {
+        for (Node* node = downcast<Node>(event.target()); node; node = node->parentInComposedTree()) {
+            if (!is<HTMLElement>(node))
+                continue;
+            auto* element = downcast<HTMLElement>(node);
+            if (element->popoverState() != PopoverState::Auto)
+                continue;
+            if (element->popoverData()->visibilityState() != PopoverVisibilityState::Showing)
+                continue;
+            return element;
+        }
+        // TODO: Handle popovertarget attributes.
+        return nullptr;
+    }();
+
+    if (event.type() == eventNames().pointerdownEvent) {
+        m_popoverPointerDownTarget = popoverToAvoidHiding;
+        return;
+    }
+
+    ASSERT(event.type() == eventNames().pointerupEvent);
+    if (m_popoverPointerDownTarget == popoverToAvoidHiding.get())
```

Comment:
The == should work without get().
```suggestion
    if (m_popoverPointerDownTarget == popoverToAvoidHiding)
```

Reply:
It seems like this doesn't work, WeakPtr can't compare against RefPtr, this is why I have to use `.get()`